tallycash/extension
# 💀 Fix default wallet toggle behavior on uniswap

⚠️ This solution is sensitive to cpu load. If the cpu load is high it can happen that change does not propagate quickly enough. In those cases usually another manual reload solves the problem.

## What

- remove window.ethereum.providers array on uniswap
  - the current [uniswap](https://github.com/Uniswap/interface/blob/main/src/connection/index.ts#L43) + [web3-react](https://github.com/Uniswap/web3-react/blob/main/packages/metamask/src/index.ts#L57) implementation looks into it for MM provider and if found caches it 
  - we don't want to force the `.isMetaMask` flag to be true unless absolutely necessary
    - if we don't set the flag we see `Injected` instead of MM which is preferable 
  - send tally config with the default wallet value on portinitializationn (instead of waiting for an actual query)
    - if we do this we can be faster than uniswap caching the provider so we can set the proper provider
 - reload uniswap on user changing the default wallet option
   - because of it's caching the proxy object change does not work 
  
## Test
- go to uniswap and change the default wallet option. Uniswap UI should reload after the change, and when pressing connect it should try to connect to the proper wallet
- go to sushiswap and change the default wallet option. It should NOT reload, but the actual change should happen

## Demo

![CleanShot 2022-07-13 at 22 38 34](https://user-images.githubusercontent.com/7690112/178830297-450af613-c320-4036-a381-d68450e65ef6.gif)
